components.md

```diff
@@ -0,0 +1,102 @@
+# What happens when the 3scale components fail
+
+This section explains what features stop working when each of the pods of the
+3scale installation are not available.
+
+## apicast-production
+
+APIcast proxies calls to the APIs configured in 3scale. If `apicast-production`
+goes down, production traffic stops working.
+
+## apicast-staging
+
+If `apicast-staging` goes down, staging traffic stops working.
+
+## backend-cron
+
+When a `backend-worker` job fails, it is pushed to a "failed jobs" queue so that
+it can be retried later. Jobs can fail, for example, when there's a Redis
+timeout. `backend-cron` is responsible for rescheduling those failed jobs. That
+means that while `backend-cron` is down, those jobs will not be rescheduled. If
+the 3scale installation is working correctly, the failed jobs queues will be
+empty almost at all times, so `backend-cron` being down is not critical.
+
+`backend-cron` is also responsible for deleting the stats of services that have
+been removed. This is run every 24h. If `backend-cron` crashes in the middle of
+the delete process, it will just continue the next time it runs.
+
+## backend-listener
+
+This is the component responsible for authorizing and rate-limiting requests.
+When it goes down, APIcast is not be able to tell whether a request should be
+authorized or not and simply denies everything. That means that when
+`backend-listener` is down, the APIs configured using 3scale are down too. This
+can be mitigated by using the APIcast caching policy, but before using it, make
+sure to check whether the trade-offs it makes suit your use case.
+
+## backend-redis
+
+It's the DB used by `backend-listener` and `backend-worker`, when it goes down,
+those 2 pods stop working.
+
+## backend-worker
+
+Processes the background jobs created by `backend-listener`. Those jobs contain
+the reported metrics. When this component goes down, the reported metrics can't
+be made effective. That means that the rate-limiting functionality loses
+accuracy because those pending reports are not taken into account. Also,
+statistics will not be up to date and the alerts and errors shown in the admin
+portal will not be triggered.
+
+## system-app
+
+If this pod goes down, many things will stop working including: admin portal UI,
+3scale APIs (accounts, analytics, etc.), developer portals, etc. Also, APIcast
+will not be able to retrieve the gateway configuration. APIcasts that are
+already running will still be able to serve traffic with the latest
+configuration that they downloaded, but new APIcast deployments will not work.
```

Comment:
Is it possible that apicast could fail in this scenario for a config that it hasn't still fetched from system (for example for an api that hasn't been hit in a while)? I know in saas it's possible, not sure with the 3scale-operator deployment method.

Reply:
Yes, you're right :+1:

Reply:
I don't remember the exact details about this.
@eloycoto can you help me on this one? In the normal 3scale deployment using the operator, if there's a call for a specific service, and more than 5 min (default value) pass, APIcast will need to fetch the config again. What happens if system is down? Does APIcast use the stale config or does it deny the request?

Reply:
Hi, 

It'll use the stale config, and no problem at all: 
https://github.com/3scale/APIcast/blob/master/gateway/src/apicast/configuration_store.lua#L99


If the PATH routing is enabled, then some bad behaviours can happen. 

Regards.

Reply:
Thanks for clarifying :+1: